Source/Moya.swift

```diff
@@ -285,3 +287,18 @@ private struct CancellableWrapper: Cancellable {
 
 /// Make the Alamofire Request type conform to our type, to prevent leaking Alamofire to plugins.
 extension Request: RequestType { }
+
+/// Make `NSURLRequest` conform to `RequestType` so that it can be sent with `PluginType` method `willSendRequest(_:target:)`.
+extension NSURLRequest: RequestType {
+    public var request: NSURLRequest? {
+      return self
+    }
+
+    public func authenticate(usingCredential credential: NSURLCredential) -> Self {
```

Comment:
Setting `startRequestsImmediately` to `false` will not work since the request will be wrapped in `NSURLSessionDataTask` after `manager.request(request)` is being called. At the time any changes to the original request would not affect the data task.